packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -125,15 +125,7 @@ export class HttpRoute extends Resource implements IHttpRoute {
       scope: this,
     });
 
-    const integration = new HttpIntegration(this, `${this.node.id}-Integration`, {
-      httpApi: props.httpApi,
-      integrationType: config.type,
-      integrationUri: config.uri,
-      method: config.method,
-      connectionId: config.connectionId,
-      connectionType: config.connectionType,
-      payloadFormatVersion: config.payloadFormatVersion,
-    });
+    const integration = props.httpApi._addIntegration(config);
```

Comment:
Interesting approach.

This implementation essentially allows the same integration subclass (i.e., subclass of `IHttpRouteIntegration`) to be used across different `HttpApi`s. 

Could there be cases where it cannot or must not be bound to another instance of `HttpApi`?
The generic philosophical question is - should it be possible for integration classes to be bound to multiple `HttpApis`.

I would've approached this differently - moved the creation of `HttpIntegration` inside the `bind()` operation and cached it.

@eladb - what do you think?

Reply:
This is actually a common issue with many of the integration classes, and generally not really addressed systemically.

If possible, it's a better design if the integration doesn't keep any internal state that prohibits reusing it with multiple `HttpApi`s, but I am aware that in some cases that's not that easy.